gpii/node_modules/lifecycleManager/src/LifecycleManager.js

```diff
@@ -107,6 +122,20 @@ var gpii = fluid.registerNamespace("gpii");
         }
     });
 
+    // triggered via events on the flowManager. Will record the fact that a logon change started (i.e. login or logout)
+    gpii.lifecycleManager.onLogonChangeStarted = function (that, userToken) {
+        that.applier.change("logonChangeInProgressFor", userToken);
+    };
+
+    // triggered via events on the flowManager. Will record the fact that a logon change has completed (i.e. login or logout)
+    gpii.lifecycleManager.onLogonChangeComplete = function (that, userToken) {
+        that.applier.change("logonChangeInProgressFor", undefined, "DELETE");
```

Comment:
Sticking these two in a transaction might guard against some race conditions in the future